reana_server/reana_admin/check_workflows.py

```diff
@@ -0,0 +1,218 @@
+    return True, "No pods are in NotReady state."
+
+
+validation_map = {
+    RunStatus.queued: [
+        _message_is_in_scheduler_queue,
+    ],
+    RunStatus.pending: [
+        _pods_exist,
+        partial(_all_batch_pods_have_phase, phase="Pending"),
+    ],
+    RunStatus.running: [
+        _pods_exist,
+        partial(_all_batch_pods_have_phase, phase="Running"),
+        _no_batch_pods_are_in_notready_state,
+    ],
+    RunStatus.finished: [
+        _pods_dont_exist,
+    ],
+    RunStatus.failed: [
+        _pods_dont_exist,
+    ],
+}
+
+
+def check_workflows(  # noqa: C901
+    date_start: Optional[datetime.datetime], date_end: Optional[datetime.datetime]
+) -> None:
+    """Check if selected workflows are in sync with database according to predefined rules."""
+    # filter workflows that need to be checked
+    statuses_to_check = validation_map.keys()
+    query_filter = [Workflow.status.in_(statuses_to_check)]
+
+    if date_start:
+        query_filter.append(Workflow.created >= date_start)
+    else:
+        last_24_date_time = datetime.datetime.now() - datetime.timedelta(hours=24)
+        query_filter.append(Workflow.created >= last_24_date_time)
+
+    if date_end:
+        query_filter.append(Workflow.created < date_end)
+
+    filtered_workflows = Session.query(Workflow).filter(*query_filter)
+
+    if filtered_workflows.count() == 0:
+        click.secho(
+            "No workflows found. Please, adjust your filter conditions.",
+            fg="red",
+        )
+        sys.exit(1)
+    click.secho(
+        f"Found {filtered_workflows.count()} workflow(s). Starting to check them...",
+    )
+
+    try:
+        scheduler_messages = _collect_messages_from_scheduler_queue(filtered_workflows)
+    except Exception as error:
+        click.secho(
+            "Error is raised when collecting scheduler messages.",
+            fg="red",
+        )
+        logging.exception(error)
+        sys.exit(1)
+
+    pods = _get_all_pods()
+    if not pods:
+        click.secho(
+            "List of active pods is empty. It is required to validate worfklows. Exiting..",
+            fg="red",
+        )
+        sys.exit(1)
+
+    in_sync_workflows: List[WorkflowCheckResult] = []
+    out_of_sync_workflows: List[WorkflowCheckResult] = []
+    for workflow in filtered_workflows:
+        filtered_pods = [pod for pod in pods if str(workflow.id_) in pod.metadata.name]
+
+        checks = validation_map[workflow.status]
+        check_results: List[CheckResult] = []
+
+        for check in checks:
+            check_results.append(check(workflow, filtered_pods, scheduler_messages))
+
+        workflow_is_not_in_sync = any([not result[0] for result in check_results])
+
+        if workflow_is_not_in_sync:
+            out_of_sync_workflows.append((workflow, check_results))
+        else:
+            in_sync_workflows.append((workflow, check_results))
```

Comment:
I think that it would be enough just to print a compact list of `in_sync_workflows` ids and we could get rid of all the other detailed info, since using this script on production with larger periods of time or even last 24 hours could result in lots of workflow output. I've simply tested with 15 workflows and it was too much output to check.
Also, the most interesting part is `out_of_sync_workflows`, so I think there is not much value to check `in_sync_workflows` output? I would also suggest to try to compact the output, to trim all the new lines and to fit as much information to the screen as possible (just to prepare for large number of workflows to parse).

Reply:
I made changes. Please, check if those are good enough.

Reply:
IMO it's very compact and nice now, thanks